src/game.cpp

```diff
@@ -414,22 +417,57 @@ PointedThing getPointedThing(Client *client, v3f player_position,
 						result.node_abovesurface = np_above;
 						mindistance = distance;
 
-						hilightboxes.clear();
-
-						if (!g_settings->getBool("enable_node_highlighting")) {
-							for (std::vector<aabb3f>::const_iterator
-									i2 = boxes.begin();
-									i2 != boxes.end(); ++i2) {
-								aabb3f box = *i2;
-								box.MinEdge += npf + v3f(-d, -d, -d) - intToFloat(camera_offset, BS);
-								box.MaxEdge += npf + v3f(d, d, d) - intToFloat(camera_offset, BS);
-								hilightboxes.push_back(box);
-							}
+						selectionboxes->clear();
+						for (std::vector<aabb3f>::const_iterator
+								i2 = boxes.begin();
+								i2 != boxes.end(); ++i2) {
+							aabb3f box = *i2;
+							box.MinEdge += v3f(-d, -d, -d);
+							box.MaxEdge += v3f(d, d, d);
+							selectionboxes->push_back(box);
 						}
+						hud->setHighlightedPos(npf, camera_offset);
+						hud->setSelectionMeshScale(v3f(1.0, 1.0, 1.0));
 					}
 				}
 			} // for coords
 
+	if (selectionboxes->size() > 0) {
+		v3f pf = hud->getHighlightedPos();
+		v3s16 p = floatToInt(pf, BS);  
+		// Get selection mesh light level
+		static const v3s16 dirs[7] = {
+			v3s16( 0, 0, 0),
+			v3s16( 0, 1, 0),
+			v3s16( 0,-1, 0),
+			v3s16( 1, 0, 0),
+			v3s16(-1, 0, 0),
+			v3s16( 0, 0, 1),
+			v3s16( 0, 0,-1)
+		};
+		u16 l = 0;
+		u16 l1 = 0;
+		for (u8 i = 0; i < 7; i++) {
+			MapNode n = map.getNodeNoEx(p + dirs[i]);
+			l1 = getInteriorLight(n, -1, nodedef);
+			if (l1 > l)
+				l = l1;
+		}
+		video::SColor c = MapBlock_LightColor(255, l, 0);
+		u8 day = c.getRed();
+		u8 night = c.getGreen();
+		u32 daynight_ratio = client->getEnv().getDayNightRatio();
+		finalColorBlend(c, day, night, daynight_ratio);
+		u32 timer = porting::getTimeMs() % 5000;
```

Comment:
I feel like these periodic constants are exactly the kind of thing that should be configurable.